Applications/Spire/Source/Ui/Window.cpp

```diff
@@ -87,14 +87,31 @@ bool Window::nativeEvent(const QByteArray& eventType, void* message,
       if(!m_frame_size) {
         m_frame_size = size();
       }
-      auto parameters = reinterpret_cast<NCCALCSIZE_PARAMS*>(msg->lParam);
-      auto& requested_client_area = parameters->rgrc[0];
-      requested_client_area.right -=
-        GetSystemMetrics(SM_CXFRAME) + GetSystemMetrics(SM_CXPADDEDBORDER);
-      requested_client_area.left +=
-        GetSystemMetrics(SM_CXFRAME) + GetSystemMetrics(SM_CXPADDEDBORDER);
-      requested_client_area.bottom -=
-        GetSystemMetrics(SM_CYFRAME) + GetSystemMetrics(SM_CXPADDEDBORDER);
+      auto hwnd = reinterpret_cast<HWND>(effectiveWinId());
+      auto placement = WINDOWPLACEMENT{};
+      if (GetWindowPlacement(hwnd, &placement)) {
+        auto& rect = reinterpret_cast<NCCALCSIZE_PARAMS*>(
+          msg->lParam)->rgrc[0];
+        if(IsZoomed(hwnd) || (IsIconic(hwnd) &&
+            placement.flags & WPF_RESTORETOMAXIMIZED)) {
```

Comment:
Why deal with the minimized window?

Reply:
In this case I'm checking to see if the Window is minimized, but will be maximized when un-minimized. In this case I still want the Window to have the non-client area sized to the screen's 'working area' (excluding the task bar) size.